homeassistant/components/apcupsd/sensor.py

```diff
@@ -406,36 +429,92 @@
-    resources = config[CONF_RESOURCES]
-
-    for resource in resources:
-        if resource.upper() not in apcups_data.status:
-            _LOGGER.warning(
-                "Sensor type: %s does not appear in the APCUPSd status output",
-                resource,
-            )
+    """Import the configurations from YAML to config flows."""
+    # This is the second step of YAML config imports, first see the comments in
+    # async_setup() of __init__.py to get an idea of how we import the YAML configs.
+
+    # Here we retrieve the partial YAML configs from the special entry id, if it does
+    # not exist it means it has already been imported.
+    conf = hass.data[DOMAIN].get(SOURCE_IMPORT)
+    if conf is None:
+        return
+
+    _LOGGER.warning(
+        "Configuration of apcupsd in YAML is deprecated and will be "
+        "removed in Home Assistant 2022.12; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+
+    # Remove the artificial entry since it's no longer needed.
+    hass.data[DOMAIN].pop(SOURCE_IMPORT)
+
+    # Our config flow allows an extra field CONF_RESOURCES and will import properly as
+    # options (although not shown in UI during config setup).
+    conf[CONF_RESOURCES] = config[CONF_RESOURCES]
+
+    _LOGGER.warning(
+        "YAML configurations loaded with host %s, port %s and resources %s ",
+        conf[CONF_HOST],
+        conf[CONF_PORT],
+        conf[CONF_RESOURCES],
+    )
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+        )
+    )
+
+    return
+
 
-    entities = [
-        APCUPSdSensor(apcups_data, description)
-        for description in SENSOR_TYPES
-        if description.key in resources
-    ]
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the APCUPSd sensors from config entries."""
+    data_service: APCUPSdData = hass.data[DOMAIN][config_entry.entry_id]
+
+    # The resources from data service are in upper-case by default, but we use
+    # lower cases throughout this integration.
+    available_resources: set[str] = {k.lower() for k, _ in data_service.status.items()}
+
+    # We use user-specified resources from imported YAML config (if available) to
+    # determine whether to enable the entity by default.
+    specified_resources = None
+    if (resources := config_entry.data.get(CONF_RESOURCES)) is not None:
+        assert isinstance(resources, list)
+        specified_resources = set(resources)
+
+    entities = []
+    for resource in available_resources:
+        if resource not in SENSORS:
+            _LOGGER.warning("Invalid resource from APCUPSd: %s", resource.upper())
+            continue
+
+        description = copy.copy(SENSORS[resource])
```

Comment:
Don't copy the description. Pass an extra parameter to represent enabled entry to the entity for the imported config data instead. Setting `_attr_entity_registry_enabled_default` takes precedence over the entity description setting.